src/SchemaEditor.tsx

```diff
@@ -26,19 +31,30 @@ export const SchemaEditor: React.FC<SchemaEditorProps> = (props) => {
          schemas: [{
             uri: "https://json-schema.app/example.json", // id of the first schema
             fileMatch: ['a://b/example.json'],
-            schema: props.schema
+            schema: schema
          }]
      });
-   }, [monaco, props.schema]);
+   }, [monaco, schema]);
+   useEffect(() => {
+     if (!validationRange || !monaco) {
+       return;
+     }
+      monaco.editor.getEditors()
+        .forEach((codeEditor) => {
+          codeEditor.setSelection(validationRange)
+          codeEditor.revealRangeAtTop(validationRange, ScrollType.Smooth)
+        });
```

Comment:
For each editor set the validation range and reveal the range at the top...is this providing the auto-scroll behaviour in the editor? Having trouble reading this without looking up the monaco docs?

Reply:
That's right, yes, so this will set the validation range as selected in the editor and then scroll to it.